util/genrest/gomodel/pathtemplatematch.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gomodel
+
+import (
+	"fmt"
+	"strings"
+)
+
+// FindValuesMatching returns the the longest string prefix that matches the templates `first` and
+// `second`, and a bool indicating whether this prefix if a full match for both expressions.
+func FindValuesMatching(first, second PathTemplate) (fullMatch bool, longestMatch string, err error) {
+	one := &traverser{template: first.Flatten()}
+	two := &traverser{template: second.Flatten()}
```

Comment:
We are passing `one` and `two` to `sementsMatch()`, and they get updated there, so we need to pass by reference. Doing the dereference here rather than at the call site is fairly arbitrary, but reads cleaner to me.

(Separately, `traverser` could have turned out to be heavy weight, and passing by reference would avoid the copying.)